cpp/models/lct_secir/model.h

```diff
@@ -186,119 +176,95 @@ class Model
             dydt[LctState::template get_first_index<InfectionState::Recovered>() - 1] - dummy;
         dydt[LctState::template get_first_index<InfectionState::Recovered>()] =
             dydt[LctState::template get_first_index<InfectionState::Recovered>()] +
-            (1 - parameters.get<DeathsPerCritical>()) * dummy;
-        dydt[LctState::template get_first_index<InfectionState::Dead>()] = parameters.get<DeathsPerCritical>() * dummy;
+            (1 - params.template get<DeathsPerCritical>()) * dummy;
+        dydt[LctState::template get_first_index<InfectionState::Dead>()] =
+            params.template get<DeathsPerCritical>() * dummy;
     }
 
     /**
-     * @brief Cumulates a simulation result with subcompartments to produce a result that divides the population only into the infection states defined in InfectionState.
+     * @brief Cumulates a simulation result with subcompartments to produce a result that divides the population only
+     *   into the infection states defined in InfectionState.
      *
-     * If the model is used for simulation, we will get a result in form of a TimeSeries with infection states divided in subcompartments.
+     * If the model is used for simulation, we will get a result in form of a TimeSeries with infection states divided 
+     * in subcompartments.
      * The function calculates a TimeSeries without subcompartmens from another TimeSeries with subcompartments. 
      * This is done by summing up the numbers in the subcompartments.
      * @param[in] result result of a simulation with the model.
      * @return result of the simulation divided in the Base infection states. 
      *  Returns TimeSeries with values -1 if calculation is not possible.
      */
-    TimeSeries<ScalarType> calculate_populations(const TimeSeries<ScalarType>& result) const
+    TimeSeries<ScalarType> calculate_compartments(const TimeSeries<ScalarType>& result) const
     {
         if (!(LctState::Count == result.get_num_elements())) {
             log_error("Result does not match infectionState of the Model.");
-            TimeSeries<ScalarType> populations((int)InfectionState::Count);
-            Eigen::VectorXd wrong_size = Eigen::VectorXd::Constant((int)InfectionState::Count, -1);
-            populations.add_time_point(-1, wrong_size);
-            return populations;
+            TimeSeries<ScalarType> compartments((Eigen::Index)InfectionState::Count);
+            Eigen::VectorXd wrong_size = Eigen::VectorXd::Constant(Eigen::Index(InfectionState::Count), -1);
+            compartments.add_time_point(-1, wrong_size);
+            return compartments;
         }
-        TimeSeries<ScalarType> populations((int)InfectionState::Count);
-        Eigen::VectorXd dummy((int)InfectionState::Count);
+        TimeSeries<ScalarType> compartments((Eigen::Index)InfectionState::Count);
+        Eigen::VectorXd dummy((Eigen::Index)InfectionState::Count);
```

Comment:
How about naming this one ~Steve~ compartments, and the TimeSeries compartment_result(s) or compartment_ts, ...? Or without renaming the TS, how about (compartment(s)_)time_point, entry, ...?